Make it easier to run unit tests againts Postgres and Postgis

Currently our [CONTRIBUTING.md](https://github.com/model-bakers/model_bakery/blob/master/CONTRIBUTING.md) only covers the simplest test scenarios using SQLite and skips postgres' specific unit tests. Although we do have them being executed in our CI, it's complicated to set up the local environment to do so without proper instructions.

## Expected behavior

Update [CONTRIBUTING.md](https://github.com/model-bakers/model_bakery/blob/master/CONTRIBUTING.md) to:

1. Have an example of how to run postgres' specifc tests;
2. Have an example of how to run postgis' specifc tests;

This will definitely require people to, somehow, have postgres configured for their environment. I really think we can use **Docker** to make this easier for us.

## Actual behavior

All the errors fails without a clear message about what's happening.


## Reproduction Steps

How to reproduce this issue.

```bash
$ TEST_DB="postgresql" python -m pytest -x
============================================================================ test session starts ============================================================================
platform linux -- Python 3.8.5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/bernardo/envs/model_bakery, configfile: setup.cfg
plugins: django-3.9.0
collected 209 items                                                                                                                                                         

tests/test_baker.py E

================================================================================== ERRORS ===================================================================================
___________________________________________ ERROR at setup of TestsBakerCreatesSimpleModel.test_consider_real_django_fields_only ____________________________________________
../../.pyenv/versions/model_bakery/lib/python3.8/site-packages/pytest_django/plugin.py:513: in _django_db_marker
    request.getfixturevalue("db")
../../.pyenv/versions/model_bakery/lib/python3.8/site-packages/pytest_django/fixtures.py:105: in django_db_setup
    db_cfg = setup_databases(
../../.pyenv/versions/model_bakery/lib/python3.8/site-packages/django/test/utils.py:170: in setup_databases
    connection.creation.create_test_db(
../../.pyenv/versions/model_bakery/lib/python3.8/site-packages/django/db/backends/base/creation.py:55: in create_test_db
    self._create_test_db(verbosity, autoclobber, keepdb)
../../.pyenv/versions/model_bakery/lib/python3.8/site-packages/django/db/backends/base/creation.py:176: in _create_test_db
    with self._nodb_cursor() as cursor:
../../.pyenv/versions/3.8.5/lib/python3.8/contextlib.py:115: in __enter__
    raise RuntimeError("generator didn't yield") from None
E   RuntimeError: generator didn't yield
============================================================================= warnings summary ==============================================================================
tests/test_baker.py::TestsBakerCreatesSimpleModel::test_consider_real_django_fields_only
  /home/bernardo/.pyenv/versions/model_bakery/lib/python3.8/site-packages/django/db/backends/postgresql/base.py:304: RuntimeWarning: Normally Django will use a connection to the 'postgres' database to avoid running initialization queries against the production database when it's not needed (for example, when running tests). Django was unable to create a connection to the 'postgres' database and will use the first PostgreSQL database instead.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================== 1 warning, 1 error in 0.21s ========================================================================

```

### Versions

Python: 3.8.5
Django: 3.1.0
Model Bakery: `master` branch

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## [Unreleased](https://github.com/model-bakers/model_bakery/tree/master)
 
 ### Added
+- [dev] Add instructions and script for running `postgres` and `postgis` tests.
 
 ### Changed
 

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -47,5 +47,5 @@ sudo apt update -y && sudo apt install -y binutils libproj-dev gdal-bin
 ./postgis-tests.sh
 ```
 
-That will spn up a `docker` container with `postgresql` and `postgis` enabled and run the full test
+That will spin up a `docker` container with `postgresql` and `postgis` enabled and run the full test
 suite.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -27,3 +27,25 @@ make lint
 ```
 
 If you don't follow the step 4, your PR may fail due to `black`, `isort`, `flake8` or `pydocstyle` warnings.
+
+To run `postgresql` and `postgis` specific tests:
+
+1. [Install `docker`](https://docs.docker.com/get-docker/).
+
+2. Install the `postgis` dependencies. Follow the
+[instructions from the Django docs](https://docs.djangoproject.com/en/3.1/ref/contrib/gis/install/geolibs/):
+
+If you are on Ubuntu/Debian you run the following:
+
+```shell
+sudo apt update -y && sudo apt install -y binutils libproj-dev gdal-bin
+```
+
+3. Run the following script:
+
+```shell
+./postgis-tests.sh
+```
+
+That will spn up a `docker` container with `postgresql` and `postgis` enabled and run the full test
+suite.

--- a/postgis-tests.sh
+++ b/postgis-tests.sh
@@ -1,0 +1,23 @@
+# Set a different port than postgres' default (in case the user is already running postgres locally)
+export PGPORT=4111
+export PGUSER=postgres
+export PGPASSWORD=postgres
+export TEST_DB=postgis
+
+# Run the postgis container
+docker run --rm --name modelbakery -e POSTGRES_HOST_AUTH_METHOD=trust -p ${PGPORT}:5432 -d postgis/postgis:11-3.0
+
+# Wait a few seconds so the DB container can start up
+echo "Waiting for DB container..."
+sleep 4s
+
+# Enable all of the extensions needed on the template1 database
+docker exec modelbakery /bin/bash -c "psql template1 -c \"CREATE EXTENSION IF NOT EXISTS citext;\" -U postgres"
+docker exec modelbakery /bin/bash -c "psql template1 -c \"CREATE EXTENSION IF NOT EXISTS hstore;\" -U postgres"
+docker exec modelbakery /bin/bash -c "psql template1 -c \"CREATE EXTENSION IF NOT EXISTS postgis;\" -U postgres"
+
+# Run the tests
+python -m pytest
+
+# Spin down the postgis container
+docker stop modelbakery

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -36,7 +36,15 @@ def pytest_configure():
 
     settings.configure(
         DATABASES={
-            "default": {"ENGINE": db_engine, "NAME": db_name, "HOST": "localhost"}
+            "default": {
+                "ENGINE": db_engine,
+                "NAME": db_name,
+                "HOST": "localhost",
+                # The following DB settings are only used for `postgresql` and `postgis`
+                "PORT": os.environ.get("PGPORT", ""),
+                "USER": os.environ.get("PGUSER", ""),
+                "PASSWORD": os.environ.get("PGPASSWORD", "")
+            }
         },
         INSTALLED_APPS=installed_apps,
         LANGUAGE_CODE="en",

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -43,7 +43,7 @@ def pytest_configure():
                 # The following DB settings are only used for `postgresql` and `postgis`
                 "PORT": os.environ.get("PGPORT", ""),
                 "USER": os.environ.get("PGUSER", ""),
-                "PASSWORD": os.environ.get("PGPASSWORD", "")
+                "PASSWORD": os.environ.get("PGPASSWORD", ""),
             }
         },
         INSTALLED_APPS=installed_apps,

